dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -139,6 +139,13 @@ abstract class AbstractTypedColumn[T, U]
   def isNotNone(implicit i0: U <:< Option[_]): ThisType[T, Boolean] =
     typed(Not(equalsTo(lit(None.asInstanceOf[U])).expr))
 
+  /** True if the current expression is a fractional number and is not NaN.
+    *
+    * apache/spark
+    */
+  def isNaN(implicit i0: Fractional[U]): ThisType[T, Boolean] =
```

Comment:
@kantlove thinking about it, I feel that Fractional is not a great bound for isNaN. I looked at Spark code and seems to be primarily for Double and Floats. Also, Fractional in Scala applies to other things, like Complex numbers, that are not valid Spark types by default. I also don't think that CatalystNumeric is a great fit for it. Maybe have another type-class called `CatalystFractional` and only have implicits for Double and Float. CatalystNumeric can be a good template for you to mimic.

Reply:
@imarios Ooh, I'm not aware that there can be other valid types for `Fractional`. You're right, a `CatalystFractional` would be more suitable.
(In fact, I already implemented this and then replaced it with `Fractional` because I think it is not necessary haha :sweat_smile:)